mars/services/scheduling/worker/execution.py

```diff
@@ -211,34 +215,8 @@ async def internal_run_subtask(self, subtask: Subtask, band_name: str):
             if subtask_info.cancelling:
                 raise asyncio.CancelledError
 
-            slot_task = asyncio.create_task(slot_manager_ref.acquire_free_slot(
-                (subtask.session_id, subtask.subtask_id)))
-            slot_id = subtask_info.slot_id = yield slot_task
-            if subtask_info.cancelling:
-                raise asyncio.CancelledError
-
-            subtask_info.result.status = SubtaskStatus.running
-            run_aiotask = asyncio.create_task(
-                subtask_api.run_subtask_in_slot(band_name, slot_id, subtask))
-            yield asyncio.shield(run_aiotask)
-            if subtask_info.cancelling:
-                raise asyncio.CancelledError
+            subtask_info.result = yield self._retry_run_subtask(subtask, band_name, subtask_api)
```

Comment:
We need to make sure quota is available before every attempt, thus our tasks does not go OOM.

Reply:
Good suggestion. I will fix it.